virtualizarr/writers/icechunk.py

```diff
@@ -96,65 +112,147 @@ def write_variables_to_icechunk_group(
+    existing_array.resize(tuple(new_shape))
+    return existing_array
+
+
+def get_axis(
+    dims: list[str],
+    dim_name: Optional[str],
+) -> int:
+    if dim_name is None:
+        raise ValueError("dim_name must be provided")
+    return dims.index(dim_name)
+
+
+def _check_compatible_arrays(
+    ma: ManifestArray, existing_array: zarr.core.array.Array, append_axis: int
+):
+    manifest_api.check_combinable_zarr_arrays([ma, existing_array])
+    manifest_api.check_same_ndims([ma.ndim, existing_array.ndim])
+    arr_shapes = [ma.shape, existing_array.shape]
+    manifest_api.check_same_shapes_except_on_concat_axis(arr_shapes, append_axis)
+
+
+def _check_compatible_encodings(encoding1, encoding2):
+    for key, value in encoding1.items():
+        if key in encoding2:
+            if encoding2[key] != value:
+                raise ValueError(
+                    f"Cannot concatenate arrays with different values for encoding key {key}: {encoding2[key]} != {value}"
+                )
 
 
 def write_virtual_variable_to_icechunk(
     store: "IcechunkStore",
     group: "Group",
     name: str,
     var: Variable,
+    append_dim: Optional[str] = None,
 ) -> None:
     """Write a single virtual variable into an icechunk store"""
     ma = cast(ManifestArray, var.data)
     zarray = ma.zarray
+    mode = store.mode.str
 
-    # creates array if it doesn't already exist
-    arr = group.require_array(
-        name=name,
-        shape=zarray.shape,
-        chunk_shape=zarray.chunks,
-        dtype=encode_dtype(zarray.dtype),
-        codecs=zarray._v3_codec_pipeline(),
-        dimension_names=var.dims,
-        fill_value=zarray.fill_value,
-        # TODO fill_value?
-    )
+    dims: list[str] = cast(list[str], list(var.dims))
+    append_axis, existing_num_chunks, arr = None, None, None
+    if append_dim and append_dim not in dims:
+        raise ValueError(
+            f"append_dim {append_dim} not found in variable dimensions {dims}"
+        )
+    if mode == "a":
+        existing_array = group[name]
+        if not isinstance(existing_array, Array):
+            raise ValueError("Expected existing array to be a zarr.core.Array")
+        append_axis = get_axis(dims, append_dim)
+
+        # check if arrays can be concatenated
+        _check_compatible_arrays(ma, existing_array, append_axis)
+        _check_compatible_encodings(var.encoding, existing_array.attrs)
+
+        # determine number of existing chunks along the append axis
+        existing_num_chunks = num_chunks(
+            array=group[name],
+            axis=append_axis,
+        )
 
-    # TODO it would be nice if we could assign directly to the .attrs property
-    for k, v in var.attrs.items():
-        arr.attrs[k] = encode_zarr_attr_value(v)
+        # resize the array
+        arr = resize_array(
+            group=group,
+            name=name,
+            var=var,
+            append_axis=append_axis,
+        )
```

Comment:
This is a little weird in that the function has a return value and a side-effect, which is bad functional programming. Something like this would be a little clearer (and avoids passing in `var` only to use `var.shape`

```suggestion
        arr = group[name]

        # resize the array
        resize_array(
            arr,
            shape_to_append=var.shape,
            append_axis=append_axis,
        )
```